File: Most Frequent Interwiev Questions/javaScriptSolution/ValidAnagram.js
function isAnagram(one, two) {
    if (one.length !== two.length) {
        return false;
    }

    let firstArray = Array.from(one);
    let secondArray = Array.from(two);

    firstArray.sort();
    secondArray.sort();

    return firstArray.every((char, index) => char === secondArray[index]);
}

let firstWord = "anagram";
let secondWord = "nagaram";
console.log(isAnagram(firstWord, secondWord));

File: Most Frequent Interwiev Questions/javaSolution/ValidAnagram.java
package javaSolution;

import java.util.Arrays;
/*
Given two strings s and t, return true if t is an anagram of s, and false otherwise.

An Anagram is a word or phrase formed by rearranging the letters of a different word or phrase,
typically using all the original letters exactly once.
 */

public class ValidAnagram {
    public static void main(String[] args) {
        String firstWord = "anagram";
        String secondWord = "nagaram";
        System.out.println(isAnagram(firstWord, secondWord));
    }

    public static boolean isAnagram(String one, String two) {
        if (one.length() != two.length()) {
            return false;
        }

        char[] firstArray = one.toCharArray();
        char[] secondArray = two.toCharArray();

        Arrays.sort(firstArray);
        Arrays.sort(secondArray);

        return Arrays.equals(firstArray, secondArray);
    }
}
